src/workerd/api/streams/writable.c++

```diff
@@ -384,24 +532,44 @@ void WritableStream::serialize(jsg::Lock& js, jsg::Serializer& serializer) {
   // TODO(soon): Support JS-backed WritableStreams. Currently this only supports native streams
   //   and IdentityTransformStream, since only they are backed by WritableStreamSink.
 
-  // NOTE: We're counting on `removeSink()`, to check that the stream is not locked and other
-  //   common checks. It's important we don't modify the WritableStream before this call.
-  auto sink = removeSink(js);
-  auto encoding = sink->disownEncodingResponsibility();
-  auto wrapper = kj::heap<WritableStreamRpcAdapter>(kj::mv(sink));
+  KJ_IF_SOME(sink, getController().removeSink(js)) {
+    // NOTE: We're counting on `removeSink()`, to check that the stream is not locked and other
+    //   common checks. It's important we don't modify the WritableStream before this call.
+    auto encoding = sink->disownEncodingResponsibility();
+    auto wrapper = kj::heap<WritableStreamRpcAdapter>(kj::mv(sink));
 
-  // Make sure this stream will be revoked if the IoContext ends.
-  ioctx.addTask(wrapper->waitForCompletionOrRevoke().attach(ioctx.registerPendingEvent()));
+    // Make sure this stream will be revoked if the IoContext ends.
+    ioctx.addTask(wrapper->waitForCompletionOrRevoke().attach(ioctx.registerPendingEvent()));
 
-  auto capnpStream = ioctx.getByteStreamFactory().kjToCapnp(kj::mv(wrapper));
+    auto capnpStream = ioctx.getByteStreamFactory().kjToCapnp(kj::mv(wrapper));
 
-  externalHandler->write(
-      [capnpStream = kj::mv(capnpStream), encoding]
-      (rpc::JsValue::External::Builder builder) mutable {
-    auto ws = builder.initWritableStream();
-    ws.setByteStream(kj::mv(capnpStream));
-    ws.setEncoding(encoding);
-  });
+    externalHandler->write(
+        [capnpStream = kj::mv(capnpStream), encoding]
+        (rpc::JsValue::External::Builder builder) mutable {
+      auto ws = builder.initWritableStream();
+      ws.setByteStream(kj::mv(capnpStream));
+      ws.setEncoding(encoding);
+    });
+  } else {
```

Comment:
For review: there's obviously a fair amount of duplication here to with the previous branch. Once we're sure this is the approach that works, then we can eliminate the duplication.